### PR TITLE
Social: Fix share status endpoint call feature check

### PR DESCRIPTION
**Pull request by gmjuhasz**
Fixes https://github.com/Automattic/jetpack-reach/issues/570

## Proposed changes:

* Return early if the feature flag is off

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack-reach/issues/570
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Go to the editor and open the jetpack tab
* Pay attention to the `share-status` endpoint call in the network tab
* Turn off the feature flag, do the first 2 steps, the call should not show up
* Do the same on Simple and Atomic sites

image

### CODE REVIEW

**Review by spsiddarthan — DISMISSED** [review]
Looks good!

File: projects/js-packages/publicize-components/src/social-store/resolvers.js [hunk 1]
```diff
@@ -72,6 +73,12 @@ export function getPostShareStatus( _postId ) {
 	return async ( { dispatch, registry } ) => {
 		// Default to the current post ID if none is provided.
 		const postId = _postId || registry.select( editorStore ).getCurrentPostId();
+		const featureFlags = registry.select( socialStore ).featureFlags();
+
+		if ( ! featureFlags.useShareStatus ) {
+			dispatch( fetchPostShareStatus( postId, false ) );
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/social-store/resolvers.js]: I don't think you need to dispatch this action here. A simple return should work fine.

File: projects/js-packages/publicize-components/src/social-store/resolvers.js [hunk 2]
```diff
@@ -72,6 +73,12 @@ export function getPostShareStatus( _postId ) {
 	return async ( { dispatch, registry } ) => {
 		// Default to the current post ID if none is provided.
 		const postId = _postId || registry.select( editorStore ).getCurrentPostId();
+		const featureFlags = registry.select( socialStore ).featureFlags();
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/social-store/resolvers.js]: When accessing the same store, you can destructure `select` in that callback like `dispatch` and `registry` and then directly use `select.featureFlags()`

```ts
return async ( { dispatch, registry, select } ) => {
	const featureFlags = select.featureFlags();

	if ( ! featureFlags.useShareStatus ) {
		return;
	}

	// Rest of the logic
```

**Review by manzoorwanijk — APPROVED** [review]
Works as expected

**Pull request merged**